server/mscalendar/availability.go

```diff
@@ -124,6 +128,49 @@ func (m *mscalendar) setUserStatuses(filteredUsers store.UserIndex, schedules []
 	return utils.JSONBlock(schedules), nil
 }
 
+func (m *mscalendar) notifyUpcomingEvent(mattermostUserID string, items []remote.ScheduleItem) {
+	for _, scheduleItem := range items {
+		loc, err := time.LoadLocation(scheduleItem.Start.Timezone)
+		if err != nil {
+			m.Logger.Errorf("problem loading location: %s", err.Error())
+			continue
+		}
+		now := time.Now().In(loc)
+		start, err := time.Parse("2006-01-02T15:04:05MST", scheduleItem.Start.DateTime+scheduleItem.Start.Timezone)
+		if err != nil {
+			m.Logger.Errorf("problem parsing date: %s", err.Error())
+			continue
+		}
+		if now.Add(upcomingEventNotificationTime).After(start.Add(-upcomingEventNotificationWindow)) && now.Add(upcomingEventNotificationTime).Before(start.Add(upcomingEventNotificationWindow)) {
+			m.renderScheduleItem(mattermostUserID, scheduleItem)
+		}
+	}
+}
+
+func (m *mscalendar) renderScheduleItem(mattermostUserId string, s remote.ScheduleItem) {
+	message := "You have an upcoming event:"
+	start, err := time.Parse("2006-01-02T15:04:05MST", s.Start.DateTime+s.Start.Timezone)
+	if err != nil {
+		m.Logger.Errorf("problem parsing start date: %s", err.Error())
+		return
+	}
+
+	end, err := time.Parse("2006-01-02T15:04:05MST", s.End.DateTime+s.End.Timezone)
+
+	message = fmt.Sprintf("\n%s-%s", start.Format("15:04MST"), end.Format("15:04MST"))
```

Comment:
Unfortunate that there is not a const for `15:04MST`. Which got me thinking, instead of showing the timezone of the event in the output, I suggest we show the times in the user's timezone. The function signature here could be `renderScheduleItem(s remote.ScheduleItem, timezone string)`

Grab the user's current timezone from a call to `GetUserTimezone`, then pass it into `renderScheduleItem`. Then use it like:

```go
start := s.Start.In(timezone).Time()
end := s.End.In(timezone).Time()
message = fmt.Sprintf("\n%s-%s", start.Format(time.Kitchen), end.Format(time.Kitchen))
```